web-animations-2/Overview.bs

```diff
@@ -2353,6 +2354,142 @@ Items sorted earlier are executed before those sorted later.
 
 </ins>
 
+Animation Triggers {#triggers}
+-----------------
+
+### Overview ### {#triggers-overview}
+
+An <dfn export>animation trigger</dfn> is used to control the playback
+of its associated [=animation=] for time-driven animations.
+Like animations, [=animation triggers=] are associated with a [=timeline=]
+and are attached to an [=animation attachment range|attachment range=].
+
+Issue: Should there be any effect of triggers on scroll-driven animations?
+
+### Animation Trigger State ### {#trigger-state}
+
+An [=animation trigger=] |trigger| with [=animation trigger type=] |type|
+and an associated [=animation=] |animation|
+has an internal <dfn export lt="animation trigger state">state</dfn> that can be one of the following:
+
+<dl dfn-for="animation trigger state" dfn-type=value>
+  <dt><dfn>idle</dfn>
+  <dd>
+    The [=animation effect=] associated |animation| remains in
+    its [=animation effect/before phase=] and stays at zero [=animation/start time=].
+
+  <dt><dfn>primary</dfn>
+  <dd>
+    The ''primary'' effect defined by |type| is applied to |animation|.
+
+  <dt><dfn>inverse</dfn>
+  <dd>
+    The ''inverse'' effect defined by |type| is applied to |animation|.
+
+</dl>
+
+Issue: Do we need a formal resolution on the spec of the idle state?
+
+The [=animation trigger state=] |state| of an [=animation trigger=] |trigger|
+is controlled by the [=animation attachment range|attachment range=] it is attached to,
+given a flag |did trigger|, as follows:
+
+<dl class=switch>
+  : If |trigger|’s [=local time=] is [=unresolved=],
+  ::
+    Then |state| is ''animation trigger state/idle'', and |did trigger| is set to false.
```

Comment:
Wouldn't setting `did trigger` to false this reset `once` triggers anytime the timeline goes idle (e.g. anytime a ScrollTimeline's scroller becomes large enough that it doesn't scroll)? I think this is probably unexpected.

Note, it doesn't look like these links, e.g. `''animation trigger state/idle''` have linked to the definitions correctly.

Reply:
Interesting! I'm not sure whether this is unexpected. I guess we should open another issue for that.

Regarding the links, I'll check and fix.

Reply:
> Wouldn't setting did trigger to false this reset once triggers anytime the timeline goes idle (e.g. anytime a ScrollTimeline's scroller becomes large enough that it doesn't scroll)? I think this is probably unexpected.

Gave this another thought. I think this should be expected, but definitely worth opening an issue and discuss this explicitly.

> Note, it doesn't look like these links, e.g. ''animation trigger state/idle'' have linked to the definitions correctly.

Right, appears as though these aren't linked correctly, but I used the suggested syntax, so not sure what to do.

Reply:
The `did trigger` flag is used to differentiate between the initial idle state and the rest of the primary/inverse states going in/out of active interval. Mostly because you always want the initial state to be a non-triggered animation and the rest can be anything else defined by type: running/paused, played/reversed, etc.

I think it's OK for it to be reset on cases like timeline becoming inactive or if setting to a new trigger.
So not sure  whether the `once and only once` is actually called for.